### PR TITLE
Feature/automated generation

### PR DESCRIPTION
**Pull request by aledeg**
Instead of manually maintaining the file, there is now a tool that grab
all repositories defined in the repositories.json file and browse through
all repositories to find extensions.
To do that, I've added a container that runs only php-cli to trigger said
tool.

The next step will be to trigger that tool automatically on a regular basis
and to update the current repository when there are changes.

### CODE REVIEW

File: Docker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,3 @@
+FROM php:7.4-cli
+
+RUN apt-get update && apt-get install -y --no-install-recommends git && apt-get purge -y --auto-remove
```
Comment (Alkarex) [inline Docker/Dockerfile:3]: I would suggest dropping this Dockerfile, using `php:8-cli` directly instead in the Makefile, and take advantage of git on the client

**Comment by Alkarex** [comment]
I somehow did not see this PR until now, sorry!
Could you resolve the conflicts?

File: Docker/Dockerfile (hunk 1 above)
Comment (Alkarex) [reply Docker/Dockerfile:3]: Ah sorry, that is not possible because you use git inside Docker

Comment (Alkarex) [reply Docker/Dockerfile:3]: But maybe consider `php:8-cli-alpine` then, which is much smaller and with a much lighter build

Comment (Alkarex) [reply Docker/Dockerfile:3]: Example https://github.com/FreshRSS/FreshRSS/blob/896f9d33637ec510937bc13a34f98f669795ab49/Docker/Dockerfile-Alpine#L1-L9

**Comment by aledeg** [comment]
I will as soon as I resume my work on that. I'll keep you posted!

File: Docker/Dockerfile (hunk 1 above)
Comment (aledeg) [reply Docker/Dockerfile]: Done!

File: generate.php [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        $gitRepositories[] = $url;
+    }
+}
+
+// ---------------------------------------- //
+// Clone git repository to extract metadata //
+// ---------------------------------------- //
+foreach ($gitRepositories as $key => $gitRepository) {
+    echo 'Processing ', $gitRepository, ' repository', PHP_EOL;
+    exec("git clone --quiet --single-branch --no-tags {$gitRepository} /tmp/extensions/{$key}");
```
Comment (Alkarex) [inline generate.php:34]:
```suggestion
    exec("git clone --quiet --depth 1 --no-tags {$gitRepository} /tmp/extensions/{$key}");
```

No need to clone the whole history for that

**Comment by Alkarex** [comment]
It would be nice to keep downloaded repositories in a cache to avoid too much resource consumption.
And if the cache already exists, then use a a git fetch and git reset instead of git clone.

File: generate.php (hunk 2 above)
Comment (aledeg) [reply generate.php]: I've added the `depth` option and kept the `single-branch` also.

**Comment by aledeg** [comment]
> It would be nice to keep downloaded repositories in a cache to avoid too much resource consumption.
> And if the cache already exists, then use a a git fetch and git reset instead of git clone.

I think it would be really hard for no real gain.
My goal is to trigger the generation tool once a day automatically with github actions. I am pretty sure I cannot keep cache between executions. The idea is to have the file always up-to-date with a delay of 24 hours maximum and to avoid manual work.

**Comment by Frenzie** [comment]
With the exception of the two framagit repos, by being on GitHub they're already in cache for all intents and purposes.

@aledeg Is this focus on master and --no-tags correct for all of the extensions? One may well update the json long before tagging a release.

**Comment by aledeg** [comment]
@Frenzie You're right. Do you have any suggestion on how I could address that?
Do we keep a policy in the file description?
Do we use a single rule for all repositories?

**Comment by Frenzie** [comment]
A single rule of tags if there are any, otherwise json might do the trick. But indeed, manual policy is probably better/safer. Not quite sure.

**Comment by aledeg** [comment]
I'll look into that!

**Comment by Alkarex** [comment]
A little conflict :-)

**Comment by aledeg** [comment]
I've solved the conflict. I'll try to find a way to address @Frenzie 's concern about the tagging issue. I have some ideas but I have to dig a little bit.

**Comment by aledeg** [comment]
I need also to dig into github actions to see how I can build the list automatically. That will be fun, I am looking forward to it.

**Comment by aledeg** [comment]
Oh! shoot.  I was not ready :/
I'll make another branch.

**Comment by Alkarex** [comment]
Ah, I thought you wanted the CI in another PR :-)

**Comment by aledeg** [comment]
Don't worry. I'll make the changes in another PR :)

**Pull request merged**